.github/workflows/e2e_run_all.yaml

```diff
@@ -148,6 +150,11 @@ jobs:
           path: src/cloud-api-adaptor/install/
           retention-days: 7
 
+      - name: Define Secure Comms Test Configs
+        id: securecomms
+        run: |
+          ./src/cloud-api-adaptor/libvirt/config_libvirt.sh >> "$GITHUB_OUTPUT"
```

Comment:
I'm a bit confused by this approach, could you not just set the `secure_comms` in the matrix below rather than going via `config_libvirt.sh`?

Reply:
If we will set secure_comms in the matrix below rather than going via config_libvirt.sh we will need to change the github workflow for every test we add. I am avoiding this by using the matrix values from a file as shown here and as suggested in https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/running-variations-of-jobs-in-a-workflow#example-using-an-output-to-define-two-matrices.

Reply:
I'm just not sure why config_libvirt is doing this matrix generation? Could you do something like add a json file that has a property of the secure_com matrix and then we could read that out in this section (and potentially re-use if for the other matrix sections in future)?

Reply:
Moved to a json file to define the e2e test matrix. 
Now serving all matrix and is easily extendable.